server/models/Document.test.js

```diff
@@ -221,6 +221,41 @@ describe("#searchForTeam", () => {
     const { totalCount } = await Document.searchForTeam(team, "test");
     expect(totalCount).toBe("2");
   });
+
+  test("should return the document when searched with their previous titles", async () => {
+    const team = await buildTeam();
+    const collection = await buildCollection({ teamId: team.id });
+    const document = await buildDocument({
+      teamId: team.id,
+      collectionId: collection.id,
+      title: "test number 1",
+    });
+
+    document.title = "change";
+    await document.save();
+
+    const { totalCount } = await Document.searchForTeam(team, "test number");
+    expect(totalCount).toBe("1");
```

Comment:
We can also test that documents that _currently_ have the title come before documents that _previously_ had the title in results ordering